derive-register/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "prometheus-client-derive-register"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+darling = "0.20.10"
```

Comment:
I would like to keep `prometheus-client`'s dependency tree as lean as possible.

For users:

- Better compile time.
- Less crates to audit.

For maintainers:
- Less maintenance work (depending on how much we need to re-invent the wheel.)

It is not only `darling` as a dependency, but also the dependencies of `darling`.

How much work do you think would it be to implement this feature without `darling`?

Reply:
Probably only an hour or two? It doesn't seem too difficult

Reply:
Great. Would you mind removing `darling` from this pull request?